### PR TITLE
Allow passing custom HTTPClient to HTTPUpdate

### PR DESCRIPTION
**Pull request by MohammedNoureldin**
This enables customizing HTTP headers which adds some extra flexibility.
This does not break anything of course because this change introduces a new constructor with a new additional HTTPClient argument for HTTPUpdate class.

### CODE REVIEW

**Comment by me-no-dev** [comment]
Wouldn't it make sense to just add something like `t_httpUpdate_return update(HTTPClient& http, const String& currentVersion);` and `t_httpUpdate_return updateSpiffs(HTTPClient& http, const String& currentVersion);` which would call `handleUpdate` directly and not pass a WiFiClient (which might not be necessary when initializing the HTTPClient)?

**Comment by MohammedNoureldin** [comment]
Well, why not. I pushed this edit as you suggested.

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp [hunk 1]
```diff
@@ -58,6 +58,11 @@ HTTPUpdateResult HTTPUpdate::update(WiFiClient& client, const String& url, const
     return handleUpdate(http, currentVersion, false);
 }
 
+HTTPUpdateResult HTTPUpdate::updateSpiffs(HTTPClient& httpClient, const String& url, const String& currentVersion)
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.cpp:61]: Does not need the `url` argument :)

File: libraries/HTTPUpdate/src/HTTPUpdate.h [hunk 2]
```diff
@@ -86,6 +86,10 @@ class HTTPUpdate
 
     t_httpUpdate_return updateSpiffs(WiFiClient& client, const String& url, const String& currentVersion = "");
 
+    t_httpUpdate_return update(HTTPClient& httpClient, const String& host, uint16_t port, const String& uri = "/",
+                               const String& currentVersion = "");
+
+    t_httpUpdate_return updateSpiffs(HTTPClient &httpClient, const String &url, const String &currentVersion = "");
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.h:92]: Does not need the `url` argument :)

File: libraries/HTTPUpdate/src/HTTPUpdate.cpp [hunk 3]
```diff
@@ -68,6 +73,12 @@ HTTPUpdateResult HTTPUpdate::updateSpiffs(WiFiClient& client, const String& url,
     return handleUpdate(http, currentVersion, true);
 }
 
+HTTPUpdateResult HTTPUpdate::update(HTTPClient& httpClient, const String& host, uint16_t port, const String& uri,
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.cpp:76]: Does not need the `host`, `port` and `uri` arguments :)

File: libraries/HTTPUpdate/src/HTTPUpdate.h [hunk 4]
```diff
@@ -86,6 +86,10 @@ class HTTPUpdate
 
     t_httpUpdate_return updateSpiffs(WiFiClient& client, const String& url, const String& currentVersion = "");
 
+    t_httpUpdate_return update(HTTPClient& httpClient, const String& host, uint16_t port, const String& uri = "/",
```
Comment (me-no-dev) [inline libraries/HTTPUpdate/src/HTTPUpdate.h:89]: Does not need the `host`, `port` and `uri` arguments :)

**Comment by me-no-dev** [comment]
you forgot to remove some arguments there :)

**Comment by MohammedNoureldin** [comment]
@me-no-dev oh sorry! I did that it in parallel with other work. I removed these redundant arguments. This should be fine now.

**Comment by me-no-dev** [comment]
Thanks :) all good now!

**Pull request merged**